src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4961,48 +4962,58 @@ void handleCase(ICaseClauseOperation caseClause, BasicBlockBuilder body, BasicBl
                 switch (caseClause.CaseKind)
                 {
                     case CaseKind.SingleValue:
-                        handleEqualityCheck(((ISingleValueCaseClauseOperation)caseClause).Value);
+                            handleEqualityCheck(((ISingleValueCaseClauseOperation)caseClause).Value);
                         break;
 
                         void handleEqualityCheck(IOperation compareWith)
                         {
-                            bool leftIsNullable = ITypeSymbolHelpers.IsNullableType(operation.Value.Type);
-                            bool rightIsNullable = ITypeSymbolHelpers.IsNullableType(compareWith.Type);
-                            bool isLifted = leftIsNullable || rightIsNullable;
-
                             EvalStackFrame frame = PushStackFrame();
                             PushOperand(OperationCloner.CloneOperation(switchValue));
                             IOperation rightOperand = Visit(compareWith);
                             IOperation leftOperand = PopOperand();
 
-                            if (isLifted)
+                            if (patternMatchingOnly)
```

Comment:
>if (patternMatchingOnly) [](start = 28, length = 24)

This doesn't feels right. If caseClause has a pattern matching semantics, I would expect us to creating IPatternCaseClauseOperation node for it rather than ISingleValueCaseClauseOperation. This way other consumers can also benefit from the clarity.